### ISSUE TITLE
CC for Deaf/HoH

### ISSUE THREAD

**Issue by vcfigueroa** [robby1066/keepposted-help#15]
# Area affected
Recording/building a message

# What's the problem
Deaf and hard of hearing participants are unable to select closed captioning or add them

# Description of feature
Please include a way to auto-cap or add our own closed captions for messages

# Alternatives and workarounds
Allow custom upload videos instead of live recording via browser so that we can use 3rd party CC apps

# Additional context

**Comment by robby1066** [comment]
@vcfigueroa Thanks for this! This is actually already [on the roadmap](https://github.com/robby1066/keepposted-help/issues/7), and hearing that it's a priority for you is really helpful.

I'm thinking about a two-phase approach for this, and I'm curious how it would work for you?

*Version one*: (which could be done very quickly) is to simply allow an upload of a caption file. I'm unsure about how useful that will be though, as I'm unclear on the effort that goes into creating captions. 

Would you be able to generate a captions file to attach to a video? (From what I can tell, it would ideally be in [.vtt format](https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API))

*Version two*: which would take longer to implement, is to have auto captioning. I've investigated enough to know roughly how it would be done, but I haven't dug too deeply into it and can't commit to a time frame on this.

So the question I have for you right now is: Does version one ☝️seem like it would be helpful for you, with the understanding that you'd have to use another tool to create the caption files offline and then upload them?

**Comment by robby1066** [comment]
[working example of closed captions in a Keep Posted message](https://robby.keepposted.io/msg/117-eb160de1de89d9058fcb0b968dbbbd68)
